### PR TITLE
Refactor cow and v3 syncs

### PR DESCRIPTION
**Pull request by franzns**
solves #979

### CODE REVIEW

File: modules/actions/pool/v3/upsert-pools.ts [hunk 1]
```diff
@@ -44,16 +45,46 @@ export const upsertPools = async (
...
+                chain,
+                upsert.pool.id,
+                blockNumber,
+            );
+            return {
+                ...upsert,
+                poolDynamicData: update.poolDynamicData,
+                poolTokenDynamicData: update.poolTokenDynamicData,
+            };
+        })
```
Comment (gmbronco) [inline modules/actions/pool/v3/upsert-pools.ts:73]:
```suggestion
        .map((upsert) => applyOnchainDataUpdateV3(
                onchainData[upsert.pool.id],
                upsert.tokens,
                chain,
                upsert.pool.id,
                blockNumber,
            )
        )
```

File: modules/actions/pool/v3/upsert-pools.ts [hunk 2]
```diff
@@ -44,16 +45,46 @@ export const upsertPools = async (
...
+            const update = enrichPoolUpsertsUsd(
+                { poolDynamicData: upsert.poolDynamicData, poolTokenDynamicData: upsert.poolTokenDynamicData },
+                prices,
+            );
+            return {
+                ...upsert,
+                poolDynamicData: update.poolDynamicData,
+                poolTokenDynamicData: update.poolTokenDynamicData,
+            };
+        });
```
Comment (gmbronco) [inline modules/actions/pool/v3/upsert-pools.ts:84]:
```suggestion
        .map((upsert) => enrichPoolUpsertsUsd(
                { poolDynamicData: upsert.poolDynamicData, poolTokenDynamicData: upsert.poolTokenDynamicData },
                prices,
            ));
```

File: modules/actions/pool/v3/sync-pools.ts [hunk 3]
```diff
@@ -25,19 +25,31 @@ const syncVaultData = async (
...
+    ));
 
     // Update pools data to the database
     for (const { poolDynamicData, poolTokenDynamicData } of poolsWithUSD) {
         try {
             await prisma.prismaPoolDynamicData.update({
                 where: {
                     poolId_chain: {
-                        poolId: poolDynamicData.poolId,
-                        chain: poolDynamicData.chain,
```
Comment (gmbronco) [inline modules/actions/pool/v3/sync-pools.ts:40]: i think it's better to use the chain from the records rather then external, we are updating known db state

File: modules/actions/pool/v3/upsert-pools.ts (hunk 1 above)
Comment (franzns) [reply modules/actions/pool/v3/upsert-pools.ts:73]: that wont work

File: modules/actions/pool/v3/upsert-pools.ts (hunk 2 above)
Comment (franzns) [reply modules/actions/pool/v3/upsert-pools.ts:84]: same, need to add the original immutable part to the return value

File: modules/actions/pool/v3/sync-pools.ts (hunk 3 above)
Comment (franzns) [reply modules/actions/pool/v3/sync-pools.ts:40]: We dont have that data here. Chain is not present in the dynamicData

**Pull request merged**